### DISCUSSION TITLE
Auto-merge sometimes works on GitLab, and sometimes it doesn't

### DISCUSSION THREAD

**Discussion by liby** [renovatebot/renovate#22905] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

https://gitlab.com/renovate-bot/renovate-runner/-/blob/main/templates/renovate.gitlab-ci.yml

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Within the repository, there are two merge requests (MRs) created by Renovate. One was auto-merged, but the other one was not.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: getPr(4093) (repository=7, branch=feature/renovate-auto-merge-private-packages-non-major-updates)
DEBUG: getMR(4093) (repository=7, branch=feature/renovate-auto-merge-private-packages-non-major-updates)
DEBUG: Branch status green (repository=7, branch=feature/renovate-auto-merge-private-packages-non-major-updates)
DEBUG: setPrCache() (repository=7, branch=feature/renovate-auto-merge-private-packages-non-major-updates)
DEBUG: Pull Request #4093 does not need updating (repository=7, branch=feature/renovate-auto-merge-private-packages-non-major-updates)
DEBUG: PR is configured for automerge (repository=7, branch=feature/renovate-auto-merge-private-packages-non-major-updates)
DEBUG: setBranchCommit() (repository=7, branch=feature/renovate-auto-merge-private-packages-non-major-updates)
```

</details>

Here, I'll continue by providing two complete logs, which were triggered the most recent two times.

The first time was a manual trigger of a Renovate run, after which the branch `feature/renovate-auto-merge-non-major-updates` was automatically merged. According to our *.gitlab-ci.yml*, once a branch starting with `feature/renovate-` is merged into the `develop` branch, Renovate will be triggered to run again, which is the second time. In the log of this run, I saw no related errors, but also no branches were auto-merged, such as `feature/renovate-auto-merge-private-packages-non-major-updates`.

[IT-WORKS-auto-merge-non-major-updates.log](https://github.com/renovatebot/renovate/files/11811987/IT-WORKS-auto-merge-non-major-updates.log)
[IT-DOES-NOT-WORK.log](https://github.com/renovatebot/renovate/files/11811990/IT-DOES-NOT-WORK.log)

### Have you created a minimal reproduction repository?

n/a

**Comment by rarkins** [comment]
```
DEBUG: Branch is behind base branch and needs rebasing (repository=7, branch=feature/renovate-auto-merge-private-packages-non-major-updates)
```

**Reply by liby** [reply]
So, during the second run, Renovate will only do one thing: rebase. It will not do anything extra, such as auto-merging, correct?

**Reply by viceice** [reply]
yes, we suggest to use platform automerge

**Reply by rarkins** [reply]
I thought platform automerge was in the config somewhere. This may be a matter of not re-enabling it?

**Reply by liby** [reply]
Indeed, I enabled `"platformAutomerge": true` in the project-level *renovate.json*, but because our GitLab MR settings require an Approval. Therefore, in the global configuration of Renovate, I set the `RENOVATE_GIT_LAB_IGNORE_APPROVALS` environment variable, hoping that Renovate's MR will ignore GitLab's approval rules when being automatically merged.

**Reply by viceice** [reply]
looks like a chicken-egg issue. PR is created with automerge, but approval rule is disabling that. then we remove the rules but don't enable automerge again.

maybe just the automerge setting and approval rule removable order is wrong.
